js/main.js

```diff
@@ -31,4 +34,108 @@ $form.addEventListener('submit', function (event) {
   data.entries.unshift(entry);
   $form.reset();
   $img.src = 'images/placeholder-image-square.jpg';
+  addEntry();
 });
+
+// <li class="row">
+//   <div class="column-half">
+//     <div class="img-container">
+//       <img src="data.photoUrl" alt="">
+//     </div>
+//   </div>
+//   <div class="column-half">
+//     <h3>
+//       data.title;
+//     </h3>
+//     <p>
+//       data.notes;
+//     </p>
+//   </div>
+// </li>
+
+function createEntry(event) {
```

Comment:
So this function should not both create and append the entries to the DOM. What you want to do is have this function loop through the entries, and move the `render` functionality to a separate function that you call within the loop. For an example, I would look at the `javascript-Dom-Creation` exercise.

The Reason for this is that you want the function that renders the `lis` to only create and return the completed `li` not append it to the page.